cmd/mdatagen/internal/templates/logs.go.tmpl

```diff
@@ -15,14 +15,38 @@ import (
 )
 
 
+{{ if .Events }}
+type EventOption interface {
+	apply(plog.LogRecord)
+}
+
+type eventOptionFunc func(plog.LogRecord)
+
+func (eof eventOptionFunc) apply(lr plog.LogRecord) {
+	eof(lr)
+}
+
+func WithEventTraceId(traceID pcommon.TraceID) EventOption {
+	return eventOptionFunc(func(lr plog.LogRecord) {
+		lr.SetTraceID(traceID)
+	})
+}
+
+func WithEventSpanId(spanID pcommon.SpanID) EventOption {
+	return eventOptionFunc(func(lr plog.LogRecord) {
+		lr.SetSpanID(spanID)
+	})
+}
+{{ end }}
```

Comment:
Can we have this `WithSpanContext` and adds both into a single options? Reduces number of allocs

Reply:
Updated.